django_material_demo/cms/utils/views.py

```diff
@@ -2,27 +2,26 @@
 from operator import or_
 
 from django.db.models import Q
-from django.utils.html import conditional_escape
-from django.utils.safestring import mark_safe
 from django_filters import CharFilter, FilterSet
 from django_filters.views import FilterView
+from django.template.loader import render_to_string
 
 
-def get_html_list(arr):
+def get_html_list(items):
     """Generate a HTML unordered list from an iterable
 
     Return an empty string instead when iterable is empty
     """
-    if len(arr) == 0:
-        return ''
-
-    #TODO: replace with template
-    value_list = [conditional_escape(x) for x in arr]
-    value_list_html = mark_safe(
-        '<ul>'
-        + ''.join('<li>' + x + '</li>' for x in value_list)
-        + '</ul>')
-    return value_list_html
+    return render_to_string('data/ul.html', {'items': items})
+
+
+def get_html_image(attrs):
+    return render_to_string('data/img.html', {'attrs': attrs})
+
+
+def get_html_anchor(content, attrs):
+    ctx = {'attrs': attrs, 'content': content}
+    return render_to_string('data/a.html', ctx)
```

Comment:
I think you have fallen into a common programming trap, which is creating new functions that do not really add extra functionality, a.k.a boilerplate. Especially in dynamic languages, strings and function names are just strings. Searching `get_html_list` and `data/ul.html` will give you the same result.

`render_to_string` by itself is a good enough abstraction. And if you want to declare more clearly the list of attributes of the dictionary passed as the second parameter, you can add comment section on top of html templates.